agent/router/manager_internal_test.go

```diff
@@ -262,55 +264,73 @@ func test_reconcileServerList(maxServers int) (bool, error) {
 	return true, nil
 }
 
-// func (l *serverList) refreshServerRebalanceTimer() {
-func TestManagerInternal_refreshServerRebalanceTimer(t *testing.T) {
-	type clusterSizes struct {
-		numNodes     int
-		numServers   int
-		minRebalance time.Duration
+func TestManager_refreshServerRebalanceTimer(t *testing.T) {
+	type testCase struct {
+		servers  int
+		nodes    int
+		expected time.Duration
 	}
-	clusters := []clusterSizes{
-		{0, 3, 2 * time.Minute},
-		{1, 0, 2 * time.Minute}, // partitioned cluster
-		{1, 3, 2 * time.Minute},
-		{2, 3, 2 * time.Minute},
-		{100, 0, 2 * time.Minute}, // partitioned
-		{100, 1, 2 * time.Minute}, // partitioned
-		{100, 3, 2 * time.Minute},
-		{1024, 1, 2 * time.Minute}, // partitioned
-		{1024, 3, 2 * time.Minute}, // partitioned
-		{1024, 5, 2 * time.Minute},
-		{16384, 1, 4 * time.Minute}, // partitioned
-		{16384, 2, 2 * time.Minute}, // partitioned
-		{16384, 3, 2 * time.Minute}, // partitioned
-		{16384, 5, 2 * time.Minute},
-		{65535, 0, 2 * time.Minute}, // partitioned
-		{65535, 1, 8 * time.Minute}, // partitioned
-		{65535, 2, 3 * time.Minute}, // partitioned
-		{65535, 3, 5 * time.Minute}, // partitioned
-		{65535, 5, 3 * time.Minute}, // partitioned
-		{65535, 7, 2 * time.Minute},
-		{1000000, 1, 4 * time.Hour},     // partitioned
-		{1000000, 2, 2 * time.Hour},     // partitioned
-		{1000000, 3, 80 * time.Minute},  // partitioned
-		{1000000, 5, 50 * time.Minute},  // partitioned
-		{1000000, 11, 20 * time.Minute}, // partitioned
-		{1000000, 19, 10 * time.Minute},
+
+	testCases := []testCase{
+		{servers: 0, nodes: 1},
+		{servers: 0, nodes: 100},
+		{servers: 0, nodes: 65535},
+		{servers: 0, nodes: 1000000},
+
+		{servers: 1, nodes: 100},
+		{servers: 1, nodes: 1024},
+		{servers: 1, nodes: 8192},
+		{servers: 1, nodes: 11520},
+		{servers: 1, nodes: 16384, expected: 4*time.Minute + 16*time.Second},
+		{servers: 1, nodes: 65535, expected: 17*time.Minute + 3984375000},
+		{servers: 1, nodes: 1000000, expected: 4*time.Hour + 20*time.Minute + 25*time.Second},
+
+		{servers: 2, nodes: 100},
+		{servers: 2, nodes: 16384},
+		{servers: 2, nodes: 65535, expected: 8*time.Minute + 31992187500},
+		{servers: 2, nodes: 1000000, expected: 2*time.Hour + 10*time.Minute + 12500*time.Millisecond},
+
+		{servers: 3, nodes: 0},
+		{servers: 3, nodes: 100},
+		{servers: 3, nodes: 1024},
+		{servers: 3, nodes: 16384},
+		{servers: 3, nodes: 32768},
+		{servers: 3, nodes: 65535, expected: 5*time.Minute + 41328125000},
+		{servers: 3, nodes: 1000000, expected: 86*time.Minute + 48333333333},
+
+		{servers: 5, nodes: 0},
+		{servers: 5, nodes: 1024},
+		{servers: 5, nodes: 16384},
+		{servers: 5, nodes: 32768},
+		{servers: 5, nodes: 56000},
+		{servers: 5, nodes: 65535, expected: 3*time.Minute + 24796875000},
+		{servers: 5, nodes: 1000000, expected: 52*time.Minute + 5*time.Second},
```

Comment:
Is this expected? I was surprised that the minimum value is used until nodes exceeds 56k (or 32k for 3 servers)

I guess this is an optimization that only matters for the very largest clusters?